sys/include/shell.h

```diff
@@ -21,13 +21,23 @@
 #define SHELL_H
 
 #include <stdint.h>
+#include "periph/pm.h"
 
 #include "kernel_defines.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @brief On native, stop RIOT on EOF
+ */
+#ifndef CONFIG_SHELL_SHUTDOWN_ON_EOF
+#  ifdef CPU_NATIVE
+#    define CONFIG_SHELL_SHUTDOWN_ON_EOF 1
+#  endif
+#endif
+
```

Comment:
Hmm although this change is simpler I don't find it accurate anymore, it does not state `shutdow on EOF`, it says `shell_run` will actually only run once. I found the old approach better.

Reply:
I deemed this cleaner than messing with the inner workings of `shell_run_once()`.

How about calling it `CONFIG_SHELL_SHUTDOWN_ON_EXIT` instead?

Reply:
Ok, but shouldn't this be in `shell_run_once` then?

Reply:
Shouldn't this be done in `shell_run_forever`, otherwise code with `shell_run_forever` would have the same issue. We can then add a note to that function documentation that will explain that on native it will shutdown since stdin is closed once a EOF is received.

Reply:
I thought if someone explicitly calls `shell_run_forever()` they would explictely want the shell to run forever - whereas `shell_run()` just gives whatever is the platform default.

Reply:
The thing is that `shell_run` says its a backport alias to `shell_run_forever`...

Reply:
Ok, moved it to `shell_run_forever`.